contrib/natsio/src/test/java/com/google/cloud/dataflow/contrib/natsio/NatsIOTest.java

```diff
@@ -0,0 +1,145 @@
+		consumers = (System.getProperty("consumers") == null ? 1 :Integer.parseInt(System.getProperty("consumers")));
+		producers = (System.getProperty("producers") == null ? 1 : Integer.parseInt(System.getProperty("producers")));
+		
+		props = new Properties();
+		props.setProperty("servers", servers);
+		if (queue != null)
+			props.setProperty("queue", queue);
+	}
+	
+	
+	private BlockingDataflowPipelineOptions buildOptions() {
+		BlockingDataflowPipelineOptions options = PipelineOptionsFactory.create().as(BlockingDataflowPipelineOptions.class);
+		options.setRunner(BlockingDataflowPipelineRunner.class);
+		options.setProject(project);
+		options.setStagingLocation(stagingLocation);
+		options.setNumWorkers(1);
+		
+		return options;
+	}
+	
+	@Test
+	public void simplePublish() {
+		BlockingDataflowPipelineOptions options = buildOptions();
+		options.setJobName("simple-publish-" + System.currentTimeMillis());
+		
+		Pipeline p = Pipeline.create(options);
+		p
+			.apply(Create.of(subject)).setCoder(StringUtf8Coder.of())
+			.apply(ParDo.of(new DoFn<String, KV<String,String>>() {
+				@Override
+				public void processElement(DoFn<String, KV<String, String>>.ProcessContext c) throws Exception {
+					c.output(KV.of(c.element(), "hello world!!!"));
+				}
+			}))
+			.apply(NatsIO.Write.to(props));
+		p.run();
+	}
+	
+	@Test
+	public void publishSubscribe() throws InterruptedException {
+		// Starting a consumer
+		DataflowPipelineOptions consumerOptions = PipelineOptionsFactory.create().as(DataflowPipelineOptions.class);
+		consumerOptions.setRunner(DataflowPipelineRunner.class);
+		consumerOptions.setProject(project);
+		consumerOptions.setStagingLocation(stagingLocation);
+		consumerOptions.setNumWorkers(1);
+		consumerOptions.setJobName("consumer-" + System.currentTimeMillis());
+		
+		Pipeline p1 = Pipeline.create(consumerOptions);
+		p1
+			.apply(NatsIO.Read.withMaxNumRecords(subject, maxRecords, props))
+			.apply(ParDo.of(new DoFn<KV<String,String>,String>() {
+				@Override
+				public void processElement(DoFn<KV<String, String>, String>.ProcessContext c) throws Exception {
+					c.output(c.element().getKey() + "," + c.element().getValue());
+				}
+			}));
+		p1.run();
+
+		// Starting a producer
+		BlockingDataflowPipelineOptions producerOptions = buildOptions();
+		producerOptions.setJobName("producer-" + System.currentTimeMillis());
+		
+		Pipeline p2 = Pipeline.create(producerOptions);
+		p2
+			.apply(Create.of(subject)).setCoder(StringUtf8Coder.of())
+			.apply(ParDo.of(new DoFn<String, String>() {
+				public void processElement(DoFn<String, String>.ProcessContext c) throws Exception {
+					try {
+						Connection conn = Connection.connect(props);
+						for(int i = 0; i < loop; i++) {
+							conn.publish(subject, Integer.toString(i));
+							Thread.sleep(interval);
+						}
+					} catch (IOException e) {
+						e.printStackTrace();
```

Comment:
It looks like the desired behavior here is to ignore errors, right? (You print, but don't throw). In that case, it should be something like:

```java
LOG.error("Ignoring exception in processElement", e);
```

In general, this does not seem very informative -- it can go wrong and we don't even learn how many elements were output.